core/src/utilities/helpers/navigation-helpers.js

```diff
@@ -141,13 +150,27 @@ class NavigationHelpersClass {
       if (!arr.metaInfo) {
         arr.metaInfo = metaInfo;
       }
+      if (!arr.metaInfo.collapsible && metaInfo.collapsible) {
+        arr.metaInfo.collapsible = metaInfo.collapsible;
+      }
+      if (category && typeof category === 'object' && category.id && category.label) {
+        arr.metaInfo.label = category.label;
+        arr.metaInfo.id = category.id;
+      }
       if (!arr.metaInfo.categoryUid && key && arr.metaInfo.collapsible) {
         arr.metaInfo.categoryUid = node.parent ? this.getNodePath(node.parent) + ':' + key : key;
       }
       if (!node.hideFromNav) {
         arr.push(node);
       }
     });
+
+    Object.keys(result).forEach(category => {
+      if (result[category].metaInfo && result[category].metaInfo.id) {
+        result = renameObjKey(result, result[category].metaInfo.id, result[category].metaInfo.label);
+      }
+    });
```

Comment:
```suggestion
    Object.keys(result).forEach(category => {
      const metaInfo = result[category].metaInfo;
      if (metaInfo && metaInfo.id) {
         result[metaInfo.label] = result[metaInfo.id];
         delete result[metaInfo.id];
      }
    });
```

Reply:
It seems we can do it this way, then remove `clone` and `renameObjKey` functions on the top.
Please double-check the logic and make sure the application works as expected.

Reply:
Works 👍 . I would keep the id in metaInfo.